### PR TITLE
Use constant to look up endpoints

### PR DESCRIPTION
**Pull request by kajinamit**
... instead of hard-coding the string.

### CODE REVIEW

File: controllers/horizon_controller.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func (r *HorizonReconciler) reconcileInit(
 	//
 	// Update instance status with service endpoint url from route host information
 	//
-	instance.Status.Endpoint = apiEndpoints["public"]
+	instance.Status.Endpoint = apiEndpoints[endpoint.PublicEndpoint]
```
Comment (kajinamit) [inline controllers/horizon_controller.go:269]: Alternatively we can cast this to string but that is still redundant.

```
instance.Status.Endpoint = apiEndpoints[string(endpoint.PublicEndpoint)]
```

Comment (bshephar) [inline controllers/horizon_controller.go:269]: `endpoint.EndpointPublic`
https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/endpoint/endpoint.go#L42

**Review by bshephar — CHANGES_REQUESTED** [review]

File: controllers/horizon_controller.go (hunk 1 above)
Comment (kajinamit) [reply controllers/horizon_controller.go]: :facepalm: Fixed it. Also probably we can cast this to string explicitly for now until the lib-common change is approved.

Comment (bshephar) [reply controllers/horizon_controller.go]: ack. Do you have the link to a lib-common PR for that change? Or just being discussed atm?

**Review by bshephar — APPROVED** [review]

**Pull request merged**